### PR TITLE
gsagh-249 display ids

### PR DESCRIPTION
**Pull request by kpne**
need to add test still

### CODE REVIEW

File: GsaGH/Parameters/5_Results/VectorResultGoo.cs [hunk 1]
```diff
@@ -12,9 +12,10 @@ namespace GsaGH.Parameters
   /// </summary>
   public class VectorResultGoo : GH_GeometricGoo<GH_Vector>, IGH_PreviewData
   {
-    public readonly Point3d StartingPoint;
+    public Point3d StartingPoint;
```
Comment (DominikaLos) [inline GsaGH/Parameters/5_Results/VectorResultGoo.cs:15]: Starting Point is always the node position, and we don't want to make it possible to change it outside this class.

File: GsaGH/Parameters/5_Results/VectorResultGoo.cs [hunk 2]
```diff
@@ -12,9 +12,10 @@ namespace GsaGH.Parameters
...
     public Vector3d Direction { get; private set; }
-    public readonly IQuantity ForceValue;
+    public IQuantity ForceValue;
```
Comment (DominikaLos) [inline GsaGH/Parameters/5_Results/VectorResultGoo.cs:17]: why do we want to change it outide the class?

File: GsaGH/Parameters/5_Results/LineResultGoo.cs [hunk 3]
```diff
@@ -13,49 +13,50 @@ namespace GsaGH.Parameters
...
       _color1 = color1;
       _color2 = color2;
 
-      var gHGradient = new Grasshopper.GUI.Gradient.GH_Gradient();
-      gHGradient.AddGrip(0, _color1);
-      gHGradient.AddGrip(1, _color2);
-      PreviewResultSegments = new List<Line>();
-      PreviewResultColors = new List<Color>();
-      PreviewResultThk = new List<int>();
+      Grasshopper.GUI.Gradient.GH_Gradient gH_Gradient = new Grasshopper.GUI.Gradient.GH_Gradient();
```
Comment (DominikaLos) [inline GsaGH/Parameters/5_Results/LineResultGoo.cs:37]: here use var :)

File: GsaGH/Components/2_Geometry/ShowId.cs [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+    protected override void RegisterOutputParams(GH_OutputParamManager pManager)
+    {
+      pManager.AddPointParameter("Position", "P", "The (centre/mid) location(s) of the object(s)", GH_ParamAccess.tree);
+      pManager.HideParameter(0);
+      pManager.AddIntegerParameter("Index", "ID", "The objects ID(s)", GH_ParamAccess.tree);
+    }
+    #endregion
+
+    protected override void SolveInstance(IGH_DataAccess DA)
```
Comment (DominikaLos) [inline GsaGH/Components/2_Geometry/ShowId.cs:56]: da is an argument, should be lowerCase ;)

File: GsaGH/Components/2_Geometry/ShowId.cs [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+            if (goo is GsaElement2dGoo e2d)
+            {
+              for (int i = 0; i < e2d.Value.Mesh.Faces.Count; i++)
+              {
+                _txts.Add(e2d.Value.Ids[i].ToString());
+                ids.Append(new GH_Integer(e2d.Value.Ids[i]), path);
+                _pts.Add(e2d.Value.Mesh.Faces.GetFaceCenter(i));
+                ghPts.Append(new GH_Point(e2d.Value.Mesh.Faces.GetFaceCenter(i)), path);
+              }
+              continue;
```
Comment (DominikaLos) [inline GsaGH/Components/2_Geometry/ShowId.cs:81]: nice one :)

File: GsaGH/Components/2_Geometry/ShowId.cs [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+                {
+                  _pts.Add(resMesh.Value.Faces.GetFaceCenter(i));
+                  ghPts.Append(new GH_Point(resMesh.Value.Faces.GetFaceCenter(i)), path);
+                }
+              }
+              continue;
+            }
+
+            int id = 0;
+            Point3d pt = Point3d.Unset;
```
Comment (DominikaLos) [inline GsaGH/Components/2_Geometry/ShowId.cs:117]: use "var pt" here ;)

File: GsaGH/Parameters/5_Results/LineResultGoo.cs [hunk 7]
```diff
@@ -13,49 +13,50 @@ namespace GsaGH.Parameters
   /// </summary>
   public class LineResultGoo : GH_GeometricGoo<Line>, IGH_PreviewData
   {
-    public LineResultGoo(Line line, IQuantity result1, IQuantity result2, Color color1, Color color2, float size1, float size2) : base(line)
+    public int ElementId;
```
Comment (DominikaLos) [inline GsaGH/Parameters/5_Results/LineResultGoo.cs:16]: should be readonly - we don't want to change it after init

File: GsaGH/Parameters/5_Results/LineResultGoo.cs [hunk 8]
```diff
@@ -70,15 +71,15 @@ public override IGH_GeometricGoo Transform(Transform xform)
...
+      return new LineResultGoo(ln, Result1, Result2, _color1, _color2, _size1, _size2, ElementId);
     }
 
     public override IGH_GeometricGoo Morph(SpaceMorph xmorph)
     {
       Point3d start = xmorph.MorphPoint(Value.From);
       Point3d end = xmorph.MorphPoint(Value.To);
-      var ln = new Line(start, end);
-      return new LineResultGoo(ln, Result1, Result2, _color1, _color2, _size1, _size2);
+      Line ln = new Line(start, end);
```
Comment (DominikaLos) [inline GsaGH/Parameters/5_Results/LineResultGoo.cs:81]: var here ;)

File: GsaGH/Parameters/5_Results/LineResultGoo.cs [hunk 9]
```diff
@@ -13,49 +13,50 @@ namespace GsaGH.Parameters
   /// </summary>
   public class LineResultGoo : GH_GeometricGoo<Line>, IGH_PreviewData
   {
-    public LineResultGoo(Line line, IQuantity result1, IQuantity result2, Color color1, Color color2, float size1, float size2) : base(line)
+    public int ElementId;
+    public IQuantity Result1;
```
Comment (DominikaLos) [inline GsaGH/Parameters/5_Results/LineResultGoo.cs:17]: do we want to change it outside the class?

File: GsaGH/Parameters/5_Results/MeshResultGoo.cs [hunk 10]
```diff
@@ -1,25 +1,27 @@
...
-using System.Linq;
 
 namespace GsaGH.Parameters
 {
   public class MeshResultGoo : GH_GeometricGoo<Mesh>, IGH_PreviewData
   {
-    internal List<List<IQuantity>> ResultValues;
-    internal List<List<Point3d>> Vertices;
-
+    public List<List<IQuantity>> ResultValues = new List<List<IQuantity>>();
```
Comment (DominikaLos) [inline GsaGH/Parameters/5_Results/MeshResultGoo.cs:13]: if we don;t want to change it outside - make it readonly

File: GsaGH/Parameters/5_Results/PointResultGoo.cs [hunk 11]
```diff
@@ -9,27 +9,37 @@ namespace GsaGH.Parameters
...
-      _mSize = size;
-      _mColor = color;
+      _size = size;
+      _color = color;
+      NodeId = id;
     }
 
-    private readonly float _mSize;
-    private readonly Color _mColor;
+    public int NodeId;
```
Comment (DominikaLos) [inline GsaGH/Parameters/5_Results/PointResultGoo.cs:21]: variables, props, fields should be before constructor ;)

File: GsaGH/Parameters/5_Results/PointResultGoo.cs [hunk 12]
```diff
@@ -105,19 +121,21 @@ public override bool CastFrom(object source)
         return true;
       }
 
-      var point = Point3d.Unset;
-      if (!GH_Convert.ToPoint3d(source, ref point, GH_Conversion.Both)) return false;
-      
-      Value = point;
-      
-      return true;
+      Point3d point = Point3d.Unset;
```
Comment (DominikaLos) [inline GsaGH/Parameters/5_Results/PointResultGoo.cs:124]: var pls :)

File: GsaGHTests/Parameters/VectorResultGooTest.cs [hunk 13]
```diff
@@ -122,9 +125,9 @@ public void WhenGetTypeDescription_ThenShouldReturnValidString()
     [Fact]
     public void WhenDuplicateGeometry_ThenMethod_ShouldReturnDuplicatedObject()
     {
-      var obj = new VectorResultGoo(Point3d.Origin, Vector3d.Zero, new Force(4, ForceUnit.Kilonewton));
+      var obj = new VectorResultGoo(Point3d.Origin, Vector3d.Zero, new Force(4, ForceUnit.Kilonewton), 0);
 
-      var expectedObj = new VectorResultGoo(Point3d.Origin, Vector3d.Zero, new Force(4, ForceUnit.Kilonewton));
+      var expectedObj = new VectorResultGoo(Point3d.Origin, Vector3d.Zero, new Force(4, ForceUnit.Kilonewton), 0);
```
Comment (DominikaLos) [inline GsaGHTests/Parameters/VectorResultGooTest.cs:130]: when duplicating, also id will be the same - we need to check it

File: IntegrationTests/3_Components/ShowIdTests.cs [hunk 14]
```diff
@@ -0,0 +1,75 @@
+﻿using System;
```
Comment (DominikaLos) [inline IntegrationTests/3_Components/ShowIdTests.cs:1]: we also should have unit tests to check at least basic things. Initialisation, and fields values after creating object ;)

**Review by DominikaLos — CHANGES_REQUESTED** [review]

File: GsaGH/Parameters/5_Results/VectorResultGoo.cs (hunk 1 above)
Comment (kpne) [reply GsaGH/Parameters/5_Results/VectorResultGoo.cs:15]: users can move the results?

File: GsaGH/Parameters/5_Results/VectorResultGoo.cs (hunk 2 above)
Comment (kpne) [reply GsaGH/Parameters/5_Results/VectorResultGoo.cs:17]: users might use the scripting component to access the result IQuantity values

File: GsaGH/Parameters/5_Results/LineResultGoo.cs (hunk 3 above)
Comment (kpne) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:37]: can we do that in the refactoring you are doing?

File: GsaGH/Components/2_Geometry/ShowId.cs (hunk 4 above)
Comment (kpne) [reply GsaGH/Components/2_Geometry/ShowId.cs:56]: yep, it is DA everywhere though?

File: GsaGH/Components/2_Geometry/ShowId.cs (hunk 5 above)
Comment (kpne) [reply GsaGH/Components/2_Geometry/ShowId.cs:81]: Thanks?

File: GsaGH/Components/2_Geometry/ShowId.cs (hunk 6 above)
Comment (kpne) [reply GsaGH/Components/2_Geometry/ShowId.cs:117]: can we do this in your refactoring?

File: GsaGH/Parameters/5_Results/LineResultGoo.cs (hunk 7 above)
Comment (kpne) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:16]: users might move the results?

File: GsaGH/Parameters/5_Results/LineResultGoo.cs (hunk 8 above)
Comment (kpne) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:81]: can we do this in your refactor?

File: GsaGH/Parameters/5_Results/LineResultGoo.cs (hunk 9 above)
Comment (kpne) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:17]: users might want to access this via code from grasshopper (c# / python)

File: GsaGH/Parameters/5_Results/MeshResultGoo.cs (hunk 10 above)
Comment (kpne) [reply GsaGH/Parameters/5_Results/MeshResultGoo.cs:13]: users might want to access this via code from grasshopper (c# / python)

File: GsaGH/Parameters/5_Results/PointResultGoo.cs (hunk 11 above)
Comment (kpne) [reply GsaGH/Parameters/5_Results/PointResultGoo.cs:21]: can we move this in the next refactor round?

File: GsaGHTests/Parameters/VectorResultGooTest.cs (hunk 13 above)
Comment (kpne) [reply GsaGHTests/Parameters/VectorResultGooTest.cs:130]: yes, covered by other tests though?

File: GsaGH/Components/2_Geometry/ShowId.cs (hunk 4 above)
Comment (DominikaLos) [reply GsaGH/Components/2_Geometry/ShowId.cs:56]: yes but still it's broke google style guide

File: GsaGH/Parameters/5_Results/VectorResultGoo.cs [hunk 15]
```diff
@@ -56,17 +57,15 @@ public override BoundingBox GetBoundingBox(Transform xform)
 
     public override IGH_GeometricGoo Transform(Transform xform)
     {
-      Direction.Transform(xform);
-      this._reactionForceLine = CreateReactionForceLine(Direction);
-      return this;
+      var sPoint = StartingPoint;
+      sPoint.Transform(xform);
+      return new VectorResultGoo(sPoint, Direction, ForceValue, NodeId);
```
Comment (kpne) [inline GsaGH/Parameters/5_Results/VectorResultGoo.cs:62]: @DominikaLos FYI

File: GsaGH/Parameters/5_Results/VectorResultGoo.cs [hunk 16]
```diff
@@ -56,17 +57,15 @@ public override BoundingBox GetBoundingBox(Transform xform)
...
     }
 
     public override IGH_GeometricGoo Morph(SpaceMorph xmorph)
     {
-      var start = xmorph.MorphPoint(_reactionForceLine.From);
-      var end = xmorph.MorphPoint(_reactionForceLine.To);
-      _reactionForceLine = new Line(start, end);
-      return this;
+      var sPoint = xmorph.MorphPoint(StartingPoint);
+      return new VectorResultGoo(sPoint, Direction, ForceValue, NodeId);
```
Comment (kpne) [inline GsaGH/Parameters/5_Results/VectorResultGoo.cs:68]: @DominikaLos FYI

File: GsaGH/Parameters/5_Results/VectorResultGoo.cs (hunk 15 above)
Comment (DominikaLos) [reply GsaGH/Parameters/5_Results/VectorResultGoo.cs:62]: ok

File: GsaGH/Parameters/5_Results/VectorResultGoo.cs (hunk 16 above)
Comment (DominikaLos) [reply GsaGH/Parameters/5_Results/VectorResultGoo.cs:68]: ok

**Review by DominikaLos — APPROVED** [review]

**Review by DominikaLos — APPROVED** [review]

**Review by tlmnrnhrdt — APPROVED** [review]

**Pull request merged**